File: VB/MultipleCheckExample/Model/Task.vb
Imports System
Imports System.ComponentModel
Imports System.Linq

Namespace MultipleCheckExample.Model
	Public Class Task
		Implements INotifyPropertyChanged

		Private _Name As String
		Private _Date As Date
		Private _IsCompleted As Boolean
		Private _IsReviewed As Boolean

		Public Property Name() As String
			Get
				Return _Name
			End Get
			Set(ByVal value As String)
				If _Name <> value Then
					_Name = value
					OnPropertyChanged("Name")
				End If
			End Set
		End Property
		Public Property [Date]() As Date
			Get
				Return _Date
			End Get
			Set(ByVal value As Date)
				If _Date <> value Then
					_Date = value
					OnPropertyChanged("Date")
				End If
			End Set
		End Property
		Public Property IsCompleted() As Boolean
			Get
				Return _IsCompleted
			End Get
			Set(ByVal value As Boolean)
				If _IsCompleted <> value Then
					_IsCompleted = value
					OnPropertyChanged("IsCompleted")
				End If
			End Set
		End Property
		Public Property IsReviewed() As Boolean
			Get
				Return _IsReviewed
			End Get
			Set(ByVal value As Boolean)
				If _IsReviewed <> value Then
					_IsReviewed = value
					OnPropertyChanged("IsReviewed")
				End If
			End Set
		End Property
		Public Shared Function NewTask(Optional ByVal seed As Integer = 100) As Task
			Dim rnd As New Random(seed)
			Return New Task() With {
				.Name = "Name #" & rnd.Next(1, 100),
				.Date = New Date(2018, 3, rnd.Next(1, 30)),
				.IsCompleted = rnd.Next(0, 2) <> 0,
				.IsReviewed = rnd.Next(0, 2) <> 0
			}
		End Function
		Public Event PropertyChanged As PropertyChangedEventHandler Implements INotifyPropertyChanged.PropertyChanged
		Private Sub OnPropertyChanged(ByVal propertyName As String)
			RaiseEvent PropertyChanged(Me, New PropertyChangedEventArgs(propertyName))
		End Sub
	End Class
End Namespace

File: CS/MultipleCheckExample/Model/Task.cs
using System;
using System.ComponentModel;
using System.Linq;

namespace MultipleCheckExample.Model
{
    public class Task : INotifyPropertyChanged
    {
        private string _Name;
        private DateTime _Date;
        private bool _IsCompleted;
        private bool _IsReviewed;
        
        public string Name {
            get { return _Name; }
            set {
                if (_Name != value) {
                    _Name = value;
                    OnPropertyChanged("Name");
                }
            }
        }
        public DateTime Date {
            get { return _Date; }
            set {
                if (_Date != value) {
                    _Date = value;
                    OnPropertyChanged("Date");
                }
            }
        }
        public bool IsCompleted {
            get { return _IsCompleted; }
            set {
                if (_IsCompleted != value) {
                    _IsCompleted = value;
                    OnPropertyChanged("IsCompleted");
                }
            }
        }
        public bool IsReviewed {
            get { return _IsReviewed; }
            set {
                if (_IsReviewed != value) {
                    _IsReviewed = value;
                    OnPropertyChanged("IsReviewed");
                }
            }
        }
        public static Task NewTask(int seed = 100) {
            Random rnd = new Random(seed);
            return new Task() { Name = "Name #" + rnd.Next(1, 100), Date = new DateTime(2018, 3, rnd.Next(1, 30)), IsCompleted = rnd.Next(0, 2) != 0, IsReviewed = rnd.Next(0, 2) != 0 };
        }
        public event PropertyChangedEventHandler PropertyChanged;
        private void OnPropertyChanged(string propertyName) {
            if (PropertyChanged != null) {
                PropertyChanged(this, new PropertyChangedEventArgs(propertyName));
            }
        }
    }
}
